src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ProjectileEntityMixin.java

```diff
@@ -0,0 +1,45 @@
+package xyz.nucleoid.plasmid.mixin.game.event;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.projectile.ProjectileEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.hit.EntityHitResult;
+import net.minecraft.util.hit.HitResult;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import xyz.nucleoid.plasmid.game.GameWorld;
+import xyz.nucleoid.plasmid.game.event.BlockHitListener;
+import xyz.nucleoid.plasmid.game.event.EntityHitListener;
+
+@Mixin({ ProjectileEntity.class })
+public abstract class ProjectileEntityMixin extends Entity {
+    public ProjectileEntityMixin(EntityType<?> type, World world) {
+        super(type, world);
+    }
+
+    @Inject(method = "onCollision", at = @At("HEAD"), cancellable = true)
+    private void onCollision(HitResult hitResult, CallbackInfo ci) {
```

Comment:
Perhaps it would be better to have only one event get invoked here? Just a `ProjectileHitListener`.
It'd make the code in this method cleaner and possibly add more freedom to the listeners, however, I understand why you went with the two events approach so that games could target specifically what they need.

Reply:
Hmm, good point. Also, weird, not sure why git is showing this as a new file- this was added in a previous commit. @SuperCoder7979, thoughts on changing this?

Reply:
I thought that using 2 listeners would allow for better code, as with a singular hit listener you would need extra logic for switching between the type and casting, which would be better if it was abstracted away, in my opinion.

Reply:
That was one of my thoughts when looking it over yeah. Mostly was just looking to see what people would prefer 😛